supervisor/backups/manager.py

```diff
@@ -727,6 +727,8 @@ async def _validate_location_password(
                 raise BackupInvalidError(
                     f"Invalid password for backup {backup.slug}", _LOGGER.error
                 )
+        else:
```

Comment:
Shouldn't this be a `finally`? Otherwise we leave the password around in cache if `validate_password` raises

Reply:
We should not trust whatever is left around.

This `else` relates to `if backup.all_locations[location][ATTR_PROTECTED]:`: This way we set/clear the password on every invocation of `_validate_location_password`. We probably should rename `_validate_location_password` to `_set_location_password` or something, since it does set the password...

Reply:
oh I get it now. The collapsed view just showed the `raise` and then `else`, I thought it was a `try` - `else`, not an `if` - `else`. My bad, looks good 👍